mdm/checkin/endpoint.go

```diff
@@ -45,3 +47,25 @@ type checkinResponse struct {
 }
 
 func (r checkinResponse) error() error { return r.Err }
+
+type rejectUserAuth struct{}
+
+func (e *rejectUserAuth) Error() string {
+	return "reject user auth"
+}
+
+func (e *rejectUserAuth) UserAuthReject() bool {
+	return true
+}
+
+func (e *rejectUserAuth) error() error { return e }
```

Comment:
this method is unnecessary. `Error() string` is the one that implements the interface.

Reply:
Doesn't that need to be there because that's how it implements the `errorer` interface? Which is what in turn the `func encodeResponse` checks for?

Reply:
You're right. but that method doesn't go on the error. It is a method on the `checkinResponse`

But that brings up a good point. You should not be returning the error directly, but instead returning the checkinResponse struct, with the Err field set to your error. 

```
return checkinResponse{Err: errRejectUserAuth}, nil
```

The encode response will see that the checkinResponse is an `erroer` and pass it to the error encoding function.

Reply:
This doesn't return the error directly. All errors there are wrapped in a `checkinResponse`. However I do see that `errorer` is on the response object and not on the error itself — its superfluous here. I'll update.